lxd/storage/backend_lxd.go

```diff
@@ -3072,7 +3072,7 @@ func (b *lxdBackend) EnsureImage(fingerprint string, op *operations.Operation) e
 		// Add existing image volume's config to imgVol.
 		imgVol = b.GetVolume(drivers.VolumeTypeImage, contentType, fingerprint, imgDBVol.Config)
 
-		if b.Driver().Info().BlockBacking && imgVol.Config()["block.filesystem"] != b.poolBlockFilesystem() {
+		if b.Driver().Info().BlockBacking && imgVol.Config()["block.filesystem"] != b.poolBlockFilesystem() || !b.driver.Info().BlockBacking && imgVol.Config()["block.filesystem"] != "" {
```

Comment:
> So to check I've understood this correctly, this change is checking for if the old image volume is indicating it was using block-backed mode (because `imgVol.Config()["block.filesystem"] != "" ) and yet the pool's current `BlockBacking` mode is disabled. And so the old volume is no longer valid. Is that correct?

That is correct.

> If so, then we should avoid 2 calls to `b.Driver().Info().BlockBacking ` (and the long `if` line) by calling it one and assigning the result to a variable. And add a comment here explaining the two scenarios we are checking for.

Makes sense.